tensorflow/lite/toco/graph_transformations/hardcode_min_max.cc

```diff
@@ -63,6 +63,31 @@ bool HardcodeMinMaxForL2Normalization(Model* model, Operator* op) {
   return true;
 }
 
+bool HardcodeMinMaxForOneHot(Model* model, Operator* op) {
+  auto& output_array = model->GetArray(op->outputs[0]);
+  if (output_array.minmax) {
+    return false;
+  }
+
+  // Make sure we have OffValue minmax
+  const auto& offinput_array = model->GetArray(op->inputs[3]);
+  if (!offinput_array.minmax) {
+    return false;
+  }
+
+  // Make sure we have OnValue minmax
+  const auto& oninput_array = model->GetArray(op->inputs[2]);
+  if (!oninput_array.minmax) {
+    return false;
+  }
+  const auto& input_minmax = oninput_array.GetMinMax();
+  CHECK(!output_array.minmax);
+  auto& output_minmax = output_array.GetOrCreateMinMax();
+  output_minmax.min = input_minmax.min <= 0. ? 0. : input_minmax.min;
```

Comment:
Within this particular context (quantization scheme), you want to hardcode the output min and max values so that the range covers all possible output values. For an example, tanh's output goes from -1 to 1, so the hardcoding approximates that as close as possible (-127/128 to 1.0).

In the case of tf.one_hot, the output goes from off_value to on_value, so you ideally want to approximately hardcode the min as off_value and the max as on_value. You don't want to clamp them between 0 and 1. 

Let me double check the following with someone before you do anything: on top of all of this, you may need the computed zero point to be an integer value. In tanh, this could be why the min is set to -127/128 instead of -1. 
With that for tanh, the scale is (1 - (-127/128))/255 = 1/128, where 1 is the max and -127/128 is the min. The zero point is 0 - (-127/128)/scale  = 127, an integer as desired. If the min was -1, you'd get a zero point of 127.5.

If the above is true, it'll complicate this, and it may be easier for us to start with only supporting one_hot when off_value = 0 and on_value = 1.

Reply:
@alanchiao , thanks spending time on the PR, and providing me the review comments. I will wait for you to confirm before i do anything as suggested by you.

Regards
Amit